test/cppsuite/tests/cursor_bound_01.cpp

```diff
@@ -0,0 +1,695 @@
+              "bounded search_near found key: " + std::string(key) +
+                " with lower bound: " + lower_key + " upper bound: " + upper_key);
+            /* Assert that the range cursor has returned a key inside the bounded range. */
+            auto above_lower_key =
+              lower_key.empty() || custom_lexicographical_compare(lower_key, key, lower_inclusive);
+            auto below_upper_key =
+              upper_key.empty() || custom_lexicographical_compare(key, upper_key, upper_inclusive);
+            testutil_assert(above_lower_key && below_upper_key);
+
+            /* Decide whether the search key is inside or outside the bounded range. */
+            above_lower_key = lower_key.empty() ||
+              custom_lexicographical_compare(lower_key, search_key, lower_inclusive);
+            below_upper_key = upper_key.empty() ||
+              custom_lexicographical_compare(search_key, upper_key, upper_inclusive);
+            auto search_key_inside_range = above_lower_key && below_upper_key;
+
+            normal_cursor->set_key(normal_cursor.get(), key);
+            /* Position the normal cursor on the found key from range cursor. */
+            testutil_check(normal_cursor->search(normal_cursor.get()));
+
+            /* Call different validation methods depending on whether the search key is inside or
+             * outside the range. */
+            if (search_key_inside_range)
+                validate_successful_search_near_inside_range(
+                  normal_cursor, range_exact, search_key);
+            else {
+                testutil_assert(range_exact != 0);
+                validate_successful_search_near_outside_range(
+                  normal_cursor, lower_bound, upper_bound, above_lower_key);
+            }
+            /* Range cursor has not found anything within the set bounds. */
+        } else
+            validate_search_near_not_found(normal_cursor, lower_bound, upper_bound);
+    }
+
+    /*
+     * Validate that if the search key is inside the bounded range, that the range cursor has
+     * returned a record that is visible and is a viable record that is closest to the search key.
+     * We can use exact to perform this validation.
+     */
+    void
+    validate_successful_search_near_inside_range(
+      scoped_cursor &normal_cursor, int range_exact, const std::string &search_key)
+    {
+        int ret = 0;
+        /* Retrieve the key the normal cursor is pointing at. */
+        const char *key;
+        testutil_check(normal_cursor->get_key(normal_cursor.get(), &key));
+        logger::log_msg(LOG_TRACE,
+          "bounded search_near validating correct returned key with search key inside range as: " +
+            search_key + " and exact: " + std::to_string(range_exact));
+        /* When exact = 0, the returned key should be equal to the search key. */
+        if (range_exact == 0) {
+            testutil_assert(std::string(key).compare(search_key) == 0);
+            /*
```

Comment:
I see what you tried to do by adding this comment just before the condition `range_exact > 0` but the render is weird after `s_all`. I guess we could add the comment inside the `if` statement? Then we could be consistent with the previous `if` and the following ones.